app/routes/user/password.tsx

```diff
@@ -0,0 +1,221 @@
+  const newPassword = getFormDataString(data, 'newPassword')
+
+  invariant(oldPassword && newPassword)
+  return await updatePassword(request, oldPassword, newPassword)
+}
+
+export function WrongIdp() {
+  return (
+    <Alert type="error" heading="Error:" headingLevel="h4">
+      You must be signed in with a username and password to reset your password.
+    </Alert>
+  )
+}
+
+export function ResetPassword() {
+  const fetcher = useFetcher<typeof action>()
+  const errorMessage = fetcher.data
+
+  const [newPassword, setNewPassword] = useState('')
+  const [isOldPasswordTouched, setIsOldPasswordTouched] = useState(false)
+  const [isNewPasswordTouched, setIsNewPasswordTouched] = useState(false)
+  const [visible, setVisible] = useState(false)
+
+  const containsLower = /[a-z]/.test(newPassword)
+  const containsUpper = /[A-Z]/.test(newPassword)
+  const containsNumber = /[0-9]/.test(newPassword)
+  const validLength = newPassword.length >= 8
+  const whitespaceValid = !/^\s|\s$/.test(newPassword)
+  const containsSpecialChar = /[~`!@#$%^.&*+=\-_ [\]\\';,/{}()|\\":<>?]/g.test(
+    newPassword
+  )
+
+  const oldPasswordError = errorMessage && !isOldPasswordTouched
+  const valid =
+    whitespaceValid &&
+    containsLower &&
+    containsUpper &&
+    containsNumber &&
+    validLength &&
+    containsSpecialChar
+  const shouldDisableSubmit =
+    !newPassword || !isNewPasswordTouched || !valid || !isOldPasswordTouched
+  let isError
+  if (isNewPasswordTouched && valid) {
+    isError = false
+  }
+  if (isNewPasswordTouched && !valid) {
+    isError = true
+  }
+  const formRef = useRef<HTMLFormElement>(null)
+
+  useEffect(() => {
+    if (formRef.current && fetcher.state === 'submitting') {
+      formRef.current.reset()
+    }
+  }, [fetcher.state])
+
+  return (
+    <>
+      <h1>Reset Password</h1>
+      <>
+        <fetcher.Form method="POST" ref={formRef}>
+          <Label htmlFor="oldPassword">Old Password</Label>
+          <TextInput
+            data-focus
+            className={oldPasswordError ? 'usa-input--error' : ''}
+            name="oldPassword"
+            id="oldPassword"
+            type="password"
+            placeholder="Old Password"
+            onChange={(e) => {
+              setIsOldPasswordTouched(true)
+            }}
+          />
+          {errorMessage === 'NotAuthorizedException' && (
+            <span className="text-red">Invalid Password</span>
+          )}
+          {errorMessage === 'LimitExceededException' && (
+            <span className="text-red">
+              Attempts Exceeded. Please try again later.
+            </span>
+          )}
+          <Label htmlFor="newPassword">New Password</Label>
+          <InputGroup error={isError}>
+            <TextInput
+              name="newPassword"
+              id="newPassword"
+              className={
+                !isError && isNewPasswordTouched ? 'usa-input--success' : ''
+              }
```

Comment:
The following, I think, is a little bit more comprehensible:

```suggestion
          <InputGroup error={isNewPasswordTouched && !valid}>
            <TextInput
              name="newPassword"
              id="newPassword"
              className={
                valid ? 'usa-input--success' : ''
              }
```